kernel/src/process_standard.rs

```diff
@@ -295,6 +376,13 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
         });
     }
 
+    fn is_running(&self) -> bool {
```

Comment:
Using the verb "running" has been confusing in the past, as we probably meant for the state to be `Runnable` rather than `Running`, and it's not clear why a yielded process is running.

Is there a reason to not just stay with "is_active"?

Reply:
I'm not comfortable with adding this to the Process API, as intuitively it must always return `false`, since if the kernel is running this function the process is then not running.

I added `is_active()`, where an "active" process is one that the kernel could start running without doing any initialization on the process (such as allocating the process memory).

It seems worth it at this point to define "conditions" (or some other term not "states") that we consider Tock processes to be in, and which states map to which conditions. We keep running into this issue where we need two things: 1) precise process states for the process state machine, and 2) a more general process classification for api call validation and general process management.

However, we might find that we just need two "conditions" --> "could switch to the process if so inclined" and "cannot switch to the process without first doing some initialization/setup/verification/etc".